plugin/aws/ec2InstanceConsoleOutput.go

```diff
@@ -29,30 +26,29 @@ func newEC2InstanceConsoleOutput(inst *ec2Instance, latest bool) *ec2InstanceCon
 	} else {
 		cl.EntryBase = plugin.NewEntry("console.out")
 	}
+	cl.TurnOffCaching()
 
 	return cl
 }
 
-// TODO: Once https://github.com/puppetlabs/wash/issues/123 is resolved, create
-// a CachedConsoleLog method that we can use to generate the content + retrieve
-// the console log file's attributes (e.g. like Ctime/Mtime)
-func (cl *ec2InstanceConsoleOutput) Open(ctx context.Context) (plugin.SizedReader, error) {
-	request := &ec2Client.GetConsoleOutputInput{
-		InstanceId: awsSDK.String(cl.inst.Name()),
-	}
-	if cl.latest {
-		request.Latest = awsSDK.Bool(cl.latest)
-	}
-
-	resp, err := cl.inst.client.GetConsoleOutputWithContext(ctx, request)
+func (cl *ec2InstanceConsoleOutput) Attr(ctx context.Context) (plugin.Attributes, error) {
+	output, err := cl.inst.cachedConsoleOutput(ctx, cl.latest)
 	if err != nil {
-		return nil, err
+		return plugin.Attributes{}, err
 	}
 
-	content, err := base64.StdEncoding.DecodeString(awsSDK.StringValue(resp.Output))
+	cl.Ctime = output.mtime
+	attr, _ := cl.EntryBase.Attr(ctx)
+	attr.Size = uint64(len(output.content))
```

Comment:
Do we even need `Size` ? Could this be calculated from `SizedReader` instead (and have EntryBase mark the Size attribute as unknown by default)? Am leaning towards that. This way, it's clear that an S3 object's Ctime is its Mtime (which is precisely how S3 models its objects -- they're atomic blobs).